File: src/main/java/edu/marmara/repository/impl/CourseRepositoryImpl.java
package edu.marmara.repository.impl;

import edu.marmara.model.Course;
import edu.marmara.model.School;
import edu.marmara.repository.CourseRepository;

import java.util.ArrayList;
import java.util.Objects;

public class CourseRepositoryImpl implements CourseRepository {
    // todo: Update the corresponding json file

    @Override
    public void save(Course course) {
        School school = School.getInstance();

        if (school.getCourses() == null) {
            school.setCourses(new ArrayList<>());
        }

        school.addCourse(course);
    }

    @Override
    public Course findByCourseCode(String courseCode) {
        School school = School.getInstance();

        if (school.getCourses() == null) {
            return null;
        }

        for (Course course : school.getCourses()) {
            if (Objects.equals(course.getCourseCode(), courseCode)) {
                return course;
            }
        }

        return null;
    }
}

File: src/main/marmara/repository/impl/CourseRepositoryImpl.py
from src.main.marmara.model.School import School


class CourseRepositoryImpl:
    def save(self, course):
        school = School.get_instance()
        if school.courses is None:
            school.courses = []
        school.add_course(course)

    def find_by_course_code(self, course_code):
        school = School.get_instance()
        if school.courses is None:
            return None
        for course in school.courses:
            if course.course_code == course_code:
                return course
        return None
